File: modules/wem-api/src/main/java/com/enonic/wem/query/expr/OrderExpr.java
package com.enonic.wem.query.expr;

public abstract class OrderExpr
    implements Expression
{
    public enum Direction
    {
        ASC,
        DESC
    }

    private final Direction direction;

    public OrderExpr( final Direction direction )
    {
        this.direction = direction;
    }

    public final Direction getDirection()
    {
        return this.direction;
    }
}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/query/expr/OrderExpr.ts
module api.query.expr {

    export class OrderExpr implements Expression {

        private direction: OrderDirection;

        constructor(direction: OrderDirection) {
            this.direction = direction;
        }

        getDirection(): OrderDirection {
            return this.direction;
        }

        directionAsString(): string {
            switch (this.direction) {
            case OrderDirection.ASC:
                return "ASC";
            case OrderDirection.DESC:
                return "DESC";
            default:
                return "";
            }
        }
    }
}
